specifyweb/frontend/js_src/lib/components/SchemaConfig/UniquenessRules.tsx

```diff
@@ -0,0 +1,674 @@
+  readonly model: SpecifyModel;
+  readonly label: string;
+  readonly fields: RA<LiteralField>;
+  readonly relationships: RA<Relationship>;
+  readonly fetchedDuplicates: UniquenessRuleValidation | undefined;
+  readonly isEditing: boolean;
+  readonly onChange: (newRule: typeof rule) => void;
+  readonly onRemoved: () => void;
+}): JSX.Element {
+  const readOnly = React.useMemo(
+    () =>
+      rule.isDatabaseConstraint ||
+      !hasPermission('/schemaconfig/uniquenessrules', 'update'),
+    [rule.isDatabaseConstraint]
+  );
+
+  const [
+    isModifyingRule,
+    _setModifyingRule,
+    _setNotModifyingRule,
+    toggleModifyingRule,
+  ] = useBooleanState();
+
+  return (
+    <tr title={label}>
+      <td>
+        {fetchedDuplicates !== undefined &&
+        fetchedDuplicates.totalDuplicates > 0 ? (
+          <Button.Icon
+            icon="exclamation"
+            title=""
+            onClick={toggleModifyingRule}
+          />
+        ) : !isEditing || readOnly ? null : (
+          <Button.Small
+            aria-selected={isModifyingRule}
+            className="w-fit"
+            onClick={toggleModifyingRule}
+          >
+            {icons.pencil}
+          </Button.Small>
+        )}
+        {rule.fields.map((field, index) => (
+          <Input.Text
+            disabled
+            key={index}
+            value={
+              (
+                fields.find(({ name }) => name === field) ??
+                relationships.find(({ name }) => name === field)
+              )?.localization.name as string
+            }
+          />
+        ))}
+      </td>
+      <td>
+        <Input.Text
+          disabled
+          value={
+            rule.scopes.length === 0
+              ? schemaText.database()
+              : getFieldsFromPath(model, rule.scopes[0])
+                  .map((field) => field.localization.name as string)
+                  .join(' -> ')
+          }
+        />
+        {isModifyingRule && (
+          <ModifyUniquenessRule
+            fetchedDuplicates={fetchedDuplicates}
+            fields={fields}
+            label={label}
+            model={model}
+            readOnly={readOnly}
+            relationships={relationships}
+            rule={rule}
+            onChange={handleChanged}
+            onClose={toggleModifyingRule}
+            onRemoved={handleRemoved}
+          />
+        )}
+      </td>
+    </tr>
+  );
+}
+
+function ModifyUniquenessRule({
+  rule,
+  model,
+  readOnly,
+  label,
+  fields,
+  relationships,
+  fetchedDuplicates,
+  onChange: handleChanged,
+  onRemoved: handleRemoved,
+  onClose: handleClose,
+}: {
+  readonly rule: UniquenessRule;
+  readonly model: SpecifyModel;
+  readonly readOnly: boolean;
+  readonly label: string;
+  readonly fields: RA<LiteralField>;
+  readonly relationships: RA<Relationship>;
+  readonly fetchedDuplicates: UniquenessRuleValidation | undefined;
+  readonly onChange: (newRule: typeof rule) => void;
```

Comment:
(optional) `typeof rule` is smart. if you want to be extra smart, could do it like this:

```ts
{
  rule: [rule, setRule]
  ...
}: {
   readonly rule: GetSet<UniquenessRule>;
}
```